### ISSUE TITLE
Research response_processing_time -1

### ISSUE THREAD

**Issue by codyebberson** [medplum/medplum#2253]
Roughly 0.005% (~1 / 20,000) of API queries result in HTTP 502.  I believe this is the root cause.

From: https://repost.aws/knowledge-center/elb-alb-troubleshoot-502-errors

> **The target closed the connection with a TCP RST or a TCP FIN while the load balancer had an outstanding request to the target**
> 
> The load balancer receives a request and forwards it to the target. The target receives the request and starts to process it, but closes the connection to the load balancer too early. This usually occurs when the duration of the keep-alive timeout for the target is shorter than the idle timeout value of the load balancer. Make sure that the duration of the keep-alive timeout is greater than the idle timeout value.
> 
> Check the values for the `request_processing_time`, `target_processing_time` and `response_processing_time` fields.
> 
> See the following example access log entry:
> 
> ```
> http 2022-04-15T16:52:50.757968Z app/my-loadbalancer/50dc6c495c0c9188 192.168.131.39:2817 10.0.0.1:80 0.001 4.205 -1 502 - 94 326 "GET http://example.com:80 HTTP/1.1" "curl/7.51.0" - - arn:aws:elasticloadbalancing:us-east-1:123456789012:targetgroup/my-targets/73e2d6bc24d8a067 "Root=1-58337262-36d228ad5d99923122bbe354"
> ```
> 
> Note: In this access log entry, the request_processing_time is 0.001, the target_processing_time is 4.205, and the response_processing_time is -1.

**Comment by codyebberson** [comment]
> This usually occurs when the duration of the keep-alive timeout for the target is shorter than the idle timeout value of the load balancer.

> duration of the keep-alive timeout for the target

> shorter than

> idle timeout value of the load balancer

Node.js default keep-alive timeout is 5 seconds: https://nodejs.org/api/http.html#serverkeepalivetimeout

AWS Application Load Balancer default idle timeout is 60 seconds: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#connection-idle-timeout

This blog posts describes our situation exactly: https://shuheikagawa.com/blog/2019/04/25/keep-alive-timeout/

tl;dr: 

> Oleksii told me in a comment that only server.keepAliveTimeout was not enough on Node.js 10.15.2. It turned out that we also need to configure server.headersTimeout longer than server.keepAliveTimeout on Node.js 10.15.2 and newer. See [his issue on GitHub](https://github.com/nodejs/node/issues/27363) for more details. Thanks, Oleksii!
> 
> ```
> server.keepAliveTimeout = 61 * 1000;
> server.headersTimeout = 65 * 1000; // This should be bigger than keepAliveTimeout + your server's expected response time
> ```

**Comment by codyebberson** [comment]
This issue is actually a very interesting read: https://github.com/nodejs/node/issues/27363
